### ISSUE TITLE
azurerm_application_insights wants to replace due to change in workspace_id casing

### ISSUE THREAD

**Issue by hobbesuk** [hashicorp/terraform-provider-azurerm#18711]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

3.26.0

### Affected Resource(s)/Data Source(s)

azurerm_application_insights

### Terraform Configuration Files

```hcl
n/a
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

Nothing, the resource hasn't changed.

### Actual Behaviour

Terraform wants to replace the resource which would mean loosing all the logs

### Steps to Reproduce

terraform plan

Produces output like this:

```
  # module.web.azurerm_application_insights.web_ws["instance"] must be replaced
-/+ resource "azurerm_application_insights" "web_ws" {
      ~ app_id                                = "xxx" -> (known after apply)
      ~ connection_string                     = (sensitive value)
      ~ daily_data_cap_in_gb                  = 100 -> (known after apply)
      ~ daily_data_cap_notifications_disabled = false -> (known after apply)
      ~ id                                    = "/subscriptions/xxx/resourceGroups/canary-web/providers/Microsoft.Insights/components/canary-insights" -> (known after apply)
      ~ instrumentation_key                   = (sensitive value)
        name                                  = "canary-insights"
      ~ workspace_id                          = "/subscriptions/xxx/resourcegroups/monitoring/providers/microsoft.operationalinsights/workspaces/analytics-Canary" -> "/subscriptions/xxx/resourceGroups/monitoring/providers/Microsoft.OperationalInsights/workspaces/analytics-Canary" # forces replacement
        # (10 unchanged attributes hidden)
    }
```

Notice the change of `resoucegroups` to `resourceGroups` and `microsoft.operationalinsights` to `Microsoft.OperationalInsights`

I've seen older version of 3.x have issues around this but seems that all issues have been fixed?

Even if I go and edit the state file to change the references to what it expects and a `plan -refresh-only` is clean, but a normal `plan` wants to replace the resource.

**Comment by catriona-m** [comment]
Hi @hobbesuk thanks for opening this. I tried reproducing this but wasn't able to produce the error you saw. Is there any other information that might be relevant for reproducing?

**Comment by nnnvvvcv** [comment]
We are also seeing this behaviour. @hobbesuk did you find a workaround?

**Comment by pacorreia** [comment]
I'll probably open a new issue, but still pretty valid to refer here, that changing the workspace of an application insights should not require its replacement.

I any of us got to Application Insights properties and change the log analytics workspace to another, that an allowed operation!

So here it's wrong to trigger a replacement, worst case should give an error message on invalid workspace resource id, but never to replace application insights.

This is achieved by using the following operation: 
```
PATCH https://management.azure.com/subscriptions/<subscription_id>/resourceGroups/<resource_group_name>/providers/microsoft.insights/components/<application_insights_name>?api-version=2020-02-02-preview
Authorization: Bearer <access_token>
Content-Type: application/json

{
  "properties": {
    "WorkspaceResourceId": "/subscriptions/<subscription_id>/resourceGroups/<resource_group_name>/providers/Microsoft.OperationalInsights/workspaces/<log_analyticis_workspace_name>"
  }
}
```

Tested and working,

**Comment by mkusmiy** [comment]
looks like issue was fixed in 3.33 - at least issue does not reproduce anymore for me after upgrade from 3.27 to 3.47
https://github.com/hashicorp/terraform-provider-azurerm/pull/19325
https://github.com/hashicorp/terraform-provider-azurerm/issues/13348
